src/inputs/DateField.test.tsx

```diff
@@ -103,16 +103,21 @@ describe("DateField", () => {
     expect(onBlur).toBeCalledTimes(1);
   });
 
-  it("can select dates", async () => {
+  it("can select dates and calls onBlur", async () => {
     // Given a DateField with `jan2` as our date
     const onChange = jest.fn();
-    const r = await render(<DateField value={jan2} label="Date" onChange={onChange} />);
+    const onBlur = jest.fn();
+    const r = await render(<DateField value={jan2} label="Date" onChange={onChange} onBlur={onBlur} />);
 
     // When triggering the Date Picker
     fireEvent.focus(r.date());
     // Then the Date Picker should be shown
     expect(r.date_datePicker()).toBeTruthy();
 
+    // Fire the blur event with the date picker as the related target. This should not fire `onBlur`, but will set the proper focus state internally.
+    expect(onBlur).toBeCalledTimes(0);
```

Comment:
Should the `toBeCalledTimes(0)` happen after the `fireEvent.blur`?

Reply:
Yes, it definitely should be. Good catch.